src/pathfinder/water_regions.cpp

```diff
@@ -197,25 +197,29 @@ class WaterRegion
 
 		const size_t max_element_width = std::to_string(this->data.number_of_patches).size();
 
-		std::array<int, 16> traversability_NW{0};
-		for (auto bitIndex : SetBitIterator(this->data.edge_traversability_bits[DIAGDIR_NW])) *(traversability_NW.rbegin() + bitIndex) = 1;
-		Debug(map, 9, "    {:{}}", fmt::join(traversability_NW, " "), max_element_width);
+		std::string traversability = fmt::format("{:016b}", this->data.edge_traversability_bits[DIAGDIR_NW]);
```

Comment:
This solution no longer automatically scales with the WATER_REGION_EDGE_LENGTH. You could generate the formatting string using, well, another fmt::format call.

Reply:
Composing the format string at run-time loses the compile time format string checking.
Something like: `fmt::format("{:0{}b}", this->data.edge_traversability_bits[DIAGDIR_NW], WATER_REGION_EDGE_LENGTH);` ought to work without needing that.